extensions-core/stats/src/main/java/io/druid/query/aggregation/variance/VarianceHolder.java

```diff
@@ -0,0 +1,233 @@
+import com.google.common.primitives.Longs;
+
+import java.nio.ByteBuffer;
+import java.util.Comparator;
+
+/**
+ *
+ * Algorithm used here is copied from apache hive. This is description in GenericUDAFVariance
+ *
+ * Evaluate the variance using the algorithm described by Chan, Golub, and LeVeque in
+ * "Algorithms for computing the sample variance: analysis and recommendations"
+ * The American Statistician, 37 (1983) pp. 242--247.
+ * <p/>
+ * variance = variance1 + variance2 + n/(m*(m+n)) * pow(((m/n)*t1 - t2),2)
+ * <p/>
+ * where: - variance is sum[x-avg^2] (this is actually n times the variance)
+ * and is updated at every step. - n is the count of elements in chunk1 - m is
+ * the count of elements in chunk2 - t1 = sum of elements in chunk1, t2 =
+ * sum of elements in chunk2.
+ * <p/>
+ * This algorithm was proven to be numerically stable by J.L. Barlow in
+ * "Error analysis of a pairwise summation algorithm to compute sample variance"
+ * Numer. Math, 58 (1991) pp. 583--590
+ */
+public class VarianceHolder
+{
+  public static VarianceHolder from(ByteBuffer buffer)
+  {
+    return new VarianceHolder(buffer.getLong(), buffer.getDouble(), buffer.getDouble());
+  }
+
+  public static final Comparator<VarianceHolder> COMPARATOR = new Comparator<VarianceHolder>()
+  {
+    @Override
+    public int compare(VarianceHolder o1, VarianceHolder o2)
+    {
+      int compare = Longs.compare(o1.count, o2.count);
+      if (compare == 0) {
+        compare = Doubles.compare(o1.sum, o2.sum);
+        if (compare == 0) {
+          compare = Doubles.compare(o1.nvariance, o2.nvariance);
+        }
+      }
+      return compare;
+    }
+  };
+
+  static Object combineValues(Object lhs, Object rhs)
+  {
+    VarianceHolder holder1 = (VarianceHolder) lhs;
+    VarianceHolder holder2 = (VarianceHolder) rhs;
+
+    if (holder2.count == 0) {
+      return holder1;
+    }
+    if (holder1.count == 0) {
+      holder1.nvariance = holder2.nvariance;
+      holder1.count = holder2.count;
+      holder1.sum = holder2.sum;
+      return holder1;
+    }
+
+    final double ratio = holder1.count / (double) holder2.count;
+    final double t = holder1.sum / ratio - holder2.sum;
+
+    holder1.nvariance += holder2.nvariance + (ratio / (holder1.count + holder2.count) * t * t);
+    holder1.count += holder2.count;
+    holder1.sum += holder2.sum;
+
+    return holder1;
+  }
+
+  static int getMaxIntermediateSize()
+  {
+    return Longs.BYTES + Doubles.BYTES + Doubles.BYTES;
+  }
+
+  long count; // number of elements
+  double sum; // sum of elements
+  double nvariance; // sum[x-avg^2] (this is actually n times of the variance)
+
+  public VarianceHolder()
+  {
+  }
+
+  public void reset()
+  {
+    count = 0;
+    sum = 0;
+    nvariance = 0;
+  }
+
+  public VarianceHolder(long count, double sum, double nvariance)
+  {
+    this.count = count;
+    this.sum = sum;
+    this.nvariance = nvariance;
+  }
+
+  public VarianceHolder add(float v)
```

Comment:
instead of taking a float here, can we do the type check in here?

This will avoid creating different types of variance aggregators and we can only have 1 type